example/const_string/path.cpp

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CppAlliance/url
+//
+
+#include <boost/url/url.hpp>
+#include <boost/url/const_string.hpp>
+#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/path.hpp>
+#include <iostream>
+
+int
+main(int argc, char** argv)
+{
+    namespace urls = boost::urls;
+    namespace fs = boost::filesystem;
+
+    // Check command line arguments.
+    fs::path doc_root;
+    if(argc == 2)
+    {
+        doc_root = argv[1];
+    }
+    else
+    {
+        fs::path exec = argv[0];
+        exec = exec.filename();
+        std::cerr <<
+            "Usage: " << exec.c_str() << " <doc_root>\n"
+            "Using default values\n";
+        doc_root = fs::current_path();
+    }
+
+    // Parse a URL
+    urls::url u(
+        "https://www.boost.org/doc/libs/1_79_0/libs/json/doc/html/index%2Ehtml");
+    u.normalize_path();
+
+    // Construct path from segments
+    fs::path p = doc_root;
+    for (urls::const_string seg: u.segments()) {
+        p /= seg;
```

Comment:
hey not bad!

Reply:
From the discussions, I got the impression that the way this would be used is:

```c++
for (urls::const_string seg: u.segments(stack_allocator)) 
```

Reply:
Using the stack allocator is useful but it makes no difference in this example because all segments fit in the buffer, while the static allocator might require us to check if the segment fits in the buffer. So I thought we could just make it about `const_string` without the allocator in this example.

Reply:
It matters to this example, because we are supposed to be exploring the major use-cases. Avoiding allocations is one of those cases. So you should use the stack allocator and deal with overflow.